### PR TITLE
PIM-5130: Improve behat message

### PR DESCRIPTION
**Pull request by pierallard**
| Q                 | A
| ----------------- | ---
| Specs             | n
| Behats            | y
| Blue CI           | y
| Changelog updated | n
| Review and 2 GTM  | n
| Micro Demo (PO)   | n
| Migration script  | n
| Tech Doc          | n

# Goal of this PR
This PR is just here to improve message from behat exceptions.
Before, we had message like "I can't find the notification".
Now, you have the list of available notification, very useful when you have to debug and see the text of all the displayed notifications.

### CODE REVIEW

File: features/Context/AssertionContext.php [hunk 1]
```diff
@@ -670,13 +670,26 @@ public function iShouldSeeNotifications(TableNode $table)
         ];
 
         foreach ($table->getHash() as $data) {
-            $notification = $notificationWidget->find('css', sprintf('.dropdown-menu li>a:contains("%s")', $data['message']));
+            $notifications = $notificationWidget->findAll('css', '.dropdown-menu li>a');
```
Comment (juliensnz) [inline features/Context/AssertionContext.php]: why ?

File: features/Context/AssertionContext.php [hunk 2]
```diff
@@ -670,13 +670,26 @@ public function iShouldSeeNotifications(TableNode $table)
...
+
+            $matchingNotification = null;
+
+            foreach ($notifications as $notification) {
```
Comment (juliensnz) [inline features/Context/AssertionContext.php]: same here ?

**Comment by juliensnz** [comment]
Could you explain please ?

File: features/Context/AssertionContext.php (hunk 1 above)
Comment (juliensnz) [inline features/Context/AssertionContext.php]: we should not change that

File: features/Context/AssertionContext.php (hunk 2 above)
Comment (juliensnz) [inline features/Context/AssertionContext.php]: same here

File: features/Context/AssertionContext.php [hunk 3]
```diff
@@ -670,13 +670,26 @@ public function iShouldSeeNotifications(TableNode $table)
...
+                if (null === $matchingNotification && false !== strpos($notification->getText(), $data['message'])) {
+                    $matchingNotification = $notification;
+                }
+            }
+
+            if (!$matchingNotification) {
```
Comment (solivier) [inline features/Context/AssertionContext.php]: null !== $matchingNotification "!" is for booleans

**Comment by solivier** [comment]
GTM after comments fixed

**Pull request merged**